### PR TITLE
Adding if_command_has_response for MatterIDL template non determinism

### PR DESCRIPTION
**Pull request by brdandu**
Adding if_command_has_response to check if a reponse command exists for a given command

Github#342

### CODE REVIEW

File: src-electron/generator/helper-command.js [hunk 1]
```diff
@@ -407,6 +407,36 @@ async function if_command_fixed_length(commandId, options) {
   return options.fn(this)
 }
 
+/**
+ * If helper that checks if a reponse command is present for a given command
```
Comment (bzbarsky-apple) [inline src-electron/generator/helper-command.js:411]:
```suggestion
 * If helper that checks if a response command is present for a given command
```

File: src-electron/generator/helper-command.js [hunk 2]
```diff
@@ -407,6 +407,36 @@ async function if_command_fixed_length(commandId, options) {
...
+ * {{else}}
+ * command has no reponse
+ * {{/if_command_has_response}}
+ *
+ * @param commandName Name of the command
+ * @param replaceKeyWord Check for response command after removing the string
+ * from command name
+ * @returns content based on whether the command response was found or not
+ */
+async function if_command_has_response(commandName, replaceKeyWord, options) {
```
Comment (bzbarsky-apple) [inline src-electron/generator/helper-command.js:425]: But this ignores commands that have a response defined via responseName, no?   As in, there are commands with responses that this function returns false for....

Comment (brdandu) [reply src-electron/generator/helper-command.js:425]: That is correct. I added responseName to selectAllIncomingOrOutgoingCommandsForCluster such that helpers calling this function can have direct access to responseName which can be used in a #if helper. I have filed https://github.com/project-chip/zap/issues/402 for resolving command responses cleanly.

Comment (bzbarsky-apple) [reply src-electron/generator/helper-command.js:429]: I guess I don't understand how one would use this helper in practice.  Is the idea that you check for `responseName` first, and if falsy _then_ use this helper?

I guess this is all kinda broken, in that this has logic that needs to match logic in another repo, etc, etc...  Ah, well.

**Review by bzbarsky-apple — APPROVED** [review]

File: src-electron/generator/helper-command.js (hunk 2 above)
Comment (brdandu) [reply src-electron/generator/helper-command.js:429]: Yes one needs to check responseName first and then use this.

**Review by tecimovic — APPROVED** [review]
Should be rebased.

**Pull request merged**